File: Longest Palindrome in a String/Longest_Palindrome_in_a_String.java
class Solution {
    static String longestPalindrome(String s) {
        int n = s.length();
        int start = 0;
        int len = 0;
        for(int i = 0; i<n; i++) {
            int left = i,right = i;
            while(left >=0 && right < n && s.charAt(left) == s.charAt(right)) {
                if(right-left+1 > len) {
                    start = left;
                    len = right-left+1;
                }
                left--;
                right++;
            }
            left = i;
            right = i+1;
            while(left >=0 && right < n && s.charAt(left) == s.charAt(right)) {
                if(right-left+1 > len) {
                    start = left;
                    len = right-left+1;
                }
                left--;
                right++;
            }
        }
        return s.substring(start,start+len);
    }
}

File: Longest Palindrome in a String/Longest_Palindrome_in_a_String.py
class Solution:
    def longestPalindrome(self, s):
        n = len(s)
        start = 0
        size = 0
        for i in range(0,n):
            left = i
            right = i
            while left >=0 and right < n and s[left] == s[right]:
                if right-left+1 > size:
                    start = left
                    size = right-left+1
                
                left -=1
                right +=1
            
            left = i
            right = i+1
            while left >=0 and right < n and s[left] == s[right]:
                if right-left+1 > size:
                    start = left
                    size = right-left+1
                
                left -=1
                right +=1
        
        return s[start:start+size]
